### PR TITLE
Remove old aws security role on service port changes

### LINKED ISSUE #70421: change service port with aws nlb ，old security group rules not deleted
**What happened**:
I deployed a cluster on aws and created a loadbalance type service. When I modify the port of this service, the value of nodeport, loadbalance on aws will only add a new port in the security group, and will not delete the port that was originally opened.

**What you expected to happen**:
Delete the original opened port, only keep the new port

**How to reproduce it (as minimally and precisely as possible)**:
create a service like this：
svc.yaml
kind: Service
apiVersion: v1
metadata:
  name: my-service
  annotations:
    service.beta.kubernetes.io/aws-load-balancer-type: nlb
spec:
  selector:
    app: nginx
  ports:
  - name: internet-nlb
    nodePort: 30080
    port: 30080
    protocol: TCP
    targetPort: 80
  type: LoadBalancer

then securiy group will add two rules:

Custom TCP Rule | TCP | 30080 | 0.0.0.0/0 | kubernetes.io/rule...

Custom TCP Rule | TCP | 30080 | 172.31.0.0/16 | kubernetes.io/rule...

then I modify the service port,change nodeport to 30081,port to 30081

then securiy group will be
Custom TCP Rule | TCP | 30080 | 0.0.0.0/0 | kubernetes.io/rule...

Custom TCP Rule | TCP | 30080 | 172.31.0.0/16 | kubernetes.io/rule...

Custom TCP Rule | TCP | 30081 | 0.0.0.0/0 | kubernetes.io/rule...

Custom TCP Rule | TCP | 30081 | 172.31.0.0/16 | kubernetes.io/rule...

I think the rules contain 30080 should be removed,no server is open to 30080 now

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
v1.11.2
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
CentOS Linux 7 (Core)
- Kernel (e.g. `uname -a`):
Linux enn-2-129 3.10.0-693.el7.x86_64
- Install tools:
install k8s cluster with binary and Configuration file 
- Others:

/kind bug

### PR DESCRIPTION
**Pull request by b1zzu**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Remove old security role from aws security groups when changing service port using aws nlb

**Which issue(s) this PR fixes**:
Fixes #70421

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/sig aws

### CODE REVIEW

File: pkg/cloudprovider/providers/aws/aws_loadbalancer.go [hunk 1]
```diff
@@ -888,6 +863,53 @@ func (c *Cloud) updateInstanceSecurityGroupsForNLB(mappings []nlbPortMapping, in
...
+				continue
+			}
+			actualGroups = append(actualGroups, sg)
+		}
+
+		// Add desired groups to actual groups if it is not already so, in order to remove old security rules also
+		// from the desired groups
+		for _, desiredGroupId := range desiredGroupIds {
+
+			founded := false
```
Comment (jsafrane) [inline pkg/cloudprovider/providers/aws/aws_loadbalancer.go]: `founded` -> `found`

**Review by jsafrane — COMMENTED** [review]
I am not familiar with load balancer logic, so only style comments.

@kubernetes/sig-aws-misc, please do proper review.

**Comment by jsafrane** [comment]
lint is not happy:

```
Errors from golint:
pkg/cloudprovider/providers/aws/aws_loadbalancer.go:888:10: range var desiredGroupId should be desiredGroupID
pkg/cloudprovider/providers/aws/aws_loadbalancer.go:892:5: var actualGroupId should be actualGroupID
```

**Comment by s3than** [comment]
This would be good to have as we have ip ranges that need confirming

**Pull request closed without merging**